### PR TITLE
lookup references by assemblyID

### LINKED ISSUE #646: Create mapping from Human readable names to Ids for Google Genomics Reference API
Right now, the "names" in the table mapping (in RefAPISource) are things like EOSt9JOVhp3jkwE. What we want is "GRCh37".

### PR DESCRIPTION
**Pull request by akiezun**
fixes #646

### CODE REVIEW

**Comment by akiezun** [comment]
for @davidadamsphd

**Comment by davidadamsphd** [comment]
No large issues. A few small ones:
(1) Can you add more assembly ids for the other references we care about (and add them as other cases in the unit test).
(2) Can you file a ticket for us to migrate away from the IDs and only use the assembly IDs (I can't think of a reason why we'd want to stick with the gibberish IDs).

**Comment by akiezun** [comment]
added more assemblies

added #1082 

back to @davidadamsphd

File: src/main/java/org/broadinstitute/hellbender/engine/datasources/ReferenceAPISource.java [hunk 1]
```diff
@@ -76,10 +80,50 @@ public ReferenceAPISource( final PipelineOptions pipelineOptions, final String r
...
+        Utils.nonNull(pipelineOptions);
+        Utils.nonNull(referenceSetAssemblyID);
+        final SearchReferenceSetsRequest content = new SearchReferenceSetsRequest();
+        content.setAssemblyId(referenceSetAssemblyID);
+        try {
+            final Genomics genomicsService = createGenomicsService(pipelineOptions);
+            final SearchReferenceSetsResponse found = genomicsService.referencesets().search(content).execute();
+            final Set<String> referenceSetIds = found.getReferenceSets().stream().map(rs -> rs.getId()).collect(Collectors.toSet());
+            final Map<String, Reference> ret = new HashMap<>();
+            for (final String rId : referenceSetIds) {
```
Comment (davidadamsphd) [inline src/main/java/org/broadinstitute/hellbender/engine/datasources/ReferenceAPISource.java]: Should we instead verify that there is only one and if not, throw an error (ideally with the two options that can be specified directly)? That's be a good test for the regular + decoy problem we have.

**Comment by davidadamsphd** [comment]
I just have the one last issue. I'd be nice to have our code _not_ silently continue if there are several references found. After fixing that issue, combine your commit and merge at will.

**Comment by akiezun** [comment]
done

**Pull request merged**